File: server/encode.js
function removeSpaces(string) {

    let newstring = ""
    let onString = false

    for (charIndex in string.length) {

        const currentChar = string.substring(charIndex, charIndex)

        if (currentChar == '"' || currentChar == "'") {

            onString = !onString
        }

        if (onString || !onString && currentChar != " ") {

            newstring += currentChar
        }
    }

    return newstring
}

module.exports = {
    removeSpaces
}

File: client/encode.lua
function encode(string)

    local encodedString = ""
    local onString = false

    for charIndex = 1, string:len() do

        local currentChar = string:sub(charIndex, charIndex)

        if currentChar == '"' or currentChar == "'" then

            onString = not onString
        end

        if onString or not onString and currentChar ~= " " then

            encodedString = encodedString .. currentChar
        end
    end

    return encodedString
end
